### PR TITLE
v0.9.26 upgrade

### PR DESCRIPTION
**Pull request by rhuttman**

### CODE REVIEW

File: pallets/valve/src/lib.rs [hunk 1]
```diff
@@ -213,7 +213,7 @@ pub mod pallet {
 		pub fn open_pallet_gates(origin: OriginFor<T>) -> DispatchResult {
 			ensure_root(origin)?;
 
-			ClosedPallets::<T>::remove_all(Some(5));
+			let _ = ClosedPallets::<T>::clear(5, None);
```
Comment (rhuttman) [inline pallets/valve/src/lib.rs:216]: `remove_all` was deprecated. Clear is basically remove_all with a cursor. We pass in `None` at first since we have no cursor, and don't use the response as we don't need the cursor

File: runtime/neumann/src/lib.rs [hunk 2]
```diff
@@ -332,11 +335,11 @@ parameter_types! {
 	pub const SubAccountDeposit:  Balance = 1 * DOLLAR; // 53 bytes on-chain
 }
 
-type ForceOrigin = EnsureOneOf<
+type ForceOrigin = EitherOfDiverse<
```
Comment (rhuttman) [inline runtime/neumann/src/lib.rs:338]: `EnsureOneOf` was deprecated for `EitherOfDiverse`

File: runtime/turing/src/xcm_config.rs [hunk 3]
```diff
@@ -220,7 +222,7 @@ parameter_types! {
 	pub KarPerSecond: (AssetId, u128) = (
 		MultiLocation::new(
 			1,
-			X2(Parachain(parachains::karura::ID), GeneralKey(parachains::karura::KAR_KEY.to_vec())),
+			X2(Parachain(parachains::karura::ID), GeneralKey(WeakBoundedVec::<u8, ConstU32<32>>::force_from(parachains::karura::KAR_KEY.to_vec(), None))),
```
Comment (rhuttman) [inline runtime/turing/src/xcm_config.rs:225]: they changed `GeneralKey` to take a `WeakBoundedVec` instead of a `Vec`

File: node/src/command.rs [hunk 4]
```diff
@@ -394,7 +363,7 @@ pub fn run() -> Result<()> {
 
 				let genesis_state = with_runtime_or_err!(chain_spec, {
 					{
-						let block: Block = generate_genesis_block(&chain_spec, state_version)?;
+						let block: Block = generate_genesis_block(&**chain_spec, state_version)?;
```
Comment (rhuttman) [inline node/src/command.rs:366]: chain_spec is `& Box...` and we need the to get the thing in the Box. thus the double

File: node/src/chain_spec/neumann.rs [hunk 5]
```diff
@@ -341,7 +341,7 @@ fn testnet_genesis(
 		tokens: Default::default(),
 		technical_committee: Default::default(),
 		technical_membership: TechnicalMembershipConfig {
-			members: technical_memberships,
+			members: technical_memberships.try_into().unwrap(),
```
Comment (rhuttman) [inline node/src/chain_spec/neumann.rs:344]: this is now a `WeakBoundedVec`

File: pallets/automation-time/src/weights.rs [hunk 6]
```diff
@@ -60,41 +60,37 @@ pub trait WeightInfo {
 /// Weight functions for `pallet_automation_time`.
 pub struct AutomationWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
-	// Storage: unknown [0x3a7472616e73616374696f6e5f6c6576656c3a] (r:1 w:1)
```
Comment (rhuttman) [inline pallets/automation-time/src/weights.rs:63]: these are all gone now

**Comment by rhuttman** [comment]
This includes https://github.com/OAK-Foundation/OAK-blockchain/pull/218

File: pallets/automation-time/src/mock.rs [hunk 7]
```diff
@@ -190,6 +192,17 @@ impl<T: Config, C: frame_support::traits::ReservableCurrency<T::AccountId>>
...
 	}
+	fn delegator_bond_till_minimum(
+		delegator: &T::AccountId,
+		_: &T::AccountId,
+		account_minimum: BalanceOf<T>,
+	) -> Result<BalanceOf<T>, DispatchErrorWithPostInfo> {
+		let delegation = C::free_balance(&delegator)
+			.checked_sub(&account_minimum)
+			.ok_or(Error::<T>::InsufficientBalance)?;
+		C::reserve(delegator, delegation)?;
```
[the remark this answers is not in the document]
Comment (rhuttman) [reply pallets/automation-time/src/mock.rs:203]: he is using `checked_sub` instead, or am I missing something

File: pallets/automation-time/src/weights.rs [hunk 8]
```diff
@@ -160,16 +155,18 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 	}
 	// Storage: System Account (r:2 w:2)
+	// Storage: ParachainStaking DelegationScheduledRequests (r:1 w:0)
+	// Storage: ParachainStaking DelegatorReserveToLockMigrations (r:1 w:0)
+    // Storage: Balances Locks (r:1 w:1)
```
[the remark this answers is not in the document]
Comment (rhuttman) [reply pallets/automation-time/src/weights.rs:160]: just a github thing. I will have him redo the weights next week and see whats up first thing

File: pallets/valve/src/lib.rs (hunk 1 above)
Comment (rhuttman) [reply pallets/valve/src/lib.rs:216]: correct

File: runtime/turing/src/xcm_config.rs (hunk 3 above)
Comment (rhuttman) [reply runtime/turing/src/xcm_config.rs:225]: `GeneralKey` now takes a `WeakBoundedVec` with a "max" length of 32. I put max in quotes as the `force_from` will log the issue but do it anyway

File: pallets/automation-time/src/mock.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (justinzhou93) [reply pallets/automation-time/src/mock.rs:203]: merp missed that :(

**Review by lreesby — APPROVED** [review]
lgtm

File: pallets/automation-time/src/lib.rs [hunk 9]
```diff
@@ -1184,23 +1186,6 @@ pub mod pallet {
 			(task, <T as Config>::WeightInfo::run_auto_compound_delegated_stake_task())
 		}
 
-		fn compound_delegator_stake(
```
Comment (atomaka) [inline pallets/automation-time/src/lib.rs:1187]: For my sanity, this is unrelated to v0.9.26 upgrade and is just a parachain-staking refactor that came with the dependency upgrade?

**Review by atomaka — APPROVED** [review]

File: pallets/automation-time/src/lib.rs (hunk 9 above)
Comment (nvengal) [reply pallets/automation-time/src/lib.rs:1187]: Yes and no. The `v0.9.26` upgrade for parachain-staking included some large functional changes around funds being locked vs reserved. Rather than expose the implementation details of the locked funds outside parachain staking, I chose to move the "delegator bonds up to a minimum amount" functionality into parachain-staking.

File: node/Cargo.toml [hunk 10]
```diff
@@ -13,11 +13,7 @@ build = "build.rs"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [build-dependencies]
-substrate-build-script-utils = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.23" }
-
-[[bin]]
-name = "oak-collator"
-path = "src/main.rs"
```
Comment (nvengal) [inline node/Cargo.toml:20]: @rhuttman accidental rename?

File: node/src/command.rs [hunk 11]
```diff
@@ -67,11 +69,13 @@ impl SubstrateCli for Cli {
...
-		"OAK Collator\n\nThe command-line arguments provided first will be \
-		passed to the parachain node, while the arguments provided after -- will be passed \
-		to the relay chain node.\n\n\
-		oak-collator <parachain-args> -- <relay-chain-args>"
-			.into()
+		format!(
+			"OAK Collator\n\nThe command-line arguments provided first will be \
+			passed to the parachain node, while the arguments provided after -- will be passed \
+			to the relay chain node.\n\n\
+			{} oak-collator <parachain-args> -- <relay-chain-args>",
```
Comment (nvengal) [inline node/src/command.rs:76]: Duplicates binary name in output
```
oak-collator oak-collator <parachain-args> -- <relay-chain-args>
```

File: node/src/command.rs [hunk 12]
```diff
@@ -122,11 +126,13 @@ impl SubstrateCli for RelayChainCli {
...
```
Comment (nvengal) [inline node/src/command.rs:133]: Duplicate bin name

File: pallets/automation-time/src/weights.rs (hunk 8 above)
Comment (nvengal) [inline pallets/automation-time/src/weights.rs:160]: nit: looks like formatting got messed up by manual addition in this file.

**Review by nvengal — COMMENTED** [review]
Couple non-functional comments. Tested new parachain-staking functionality locally. No issues there. Have not yet tested migration, planning on testing the migrations on the pr that cuts the new runtime version.

**Review by nvengal — APPROVED** [review]
